2_idioms/src/main.rs

```diff
@@ -1,3 +1,452 @@
+use money::{Coin, Money};
+use product::Product;
+use std::any::TypeId;
+use trade::HasValue;
+mod trade {
+    pub trait HasValue {
+        fn value(&self) -> u64;
+    }
+}
+
+mod money {
+    use crate::trade::HasValue;
+    pub struct Amount(u32);
+    impl Into<Amount> for u32 {
```

Comment:
@ArsileLuci it's more idiomatic to define `From` implementation.

Also, you may not write this manually for such trivial cases and use `derive_more` crate instead.